src/state/kv_store.rs

```diff
@@ -0,0 +1,415 @@
+        if let Some(bytes) = self.get(&paths::HIGHEST_TC) {
+            let tc = TimeoutCertificate::deserialize(&mut &*bytes)
+                    .map_err(|err| KVGetError::DeserializeValueError{key: Key::HighestTC, source: err})?;
+            Ok(Some(tc))
+        } else {
+            Ok(None)
+        }
+    }
+
+    /* ↓↓↓ Previous Validator Set ↓↓↓ */
+
+    fn previous_validator_set(&self) -> Result<ValidatorSet, KVGetError> {
+        let previous_validator_set_bytes = 
+            ValidatorSetBytes::deserialize(
+                &mut &*self.get(&paths::PREVIOUS_VALIDATOR_SET)
+                .ok_or(KVGetError::ValueNotFound{key: Key::CommittedValidatorSet})?
+            ).map_err(|err| KVGetError::DeserializeValueError{key: Key::CommittedValidatorSet, source: err})?;
+        ValidatorSet::try_from(previous_validator_set_bytes)
+            .map_err(|err| KVGetError::Ed25519DalekError{key: Key::CommittedValidatorSet, source: err})
+    }
+
+    /* ↓↓↓ Validator Set Update Block Height ↓↓↓ */
+
+    fn validator_set_update_block_height(&self) -> Result<Option<BlockHeight>, KVGetError> {
+        if let Some(bytes) = self.get(&paths::VALIDATOR_SET_UPDATE_BLOCK_HEIGHT) {
+            let block_height = BlockHeight::deserialize(&mut &*bytes)
+                .map_err(|err| KVGetError::DeserializeValueError{key: Key::ValidatorSetUpdateHeight, source: err})?;
+            Ok(Some(block_height))
+        } else {
+            Ok(None)
+        }
+    }
+
+    /* ↓↓↓ Validator Set Update Complete ↓↓↓ */
+    fn validator_set_update_complete(&self) -> Result<bool, KVGetError> {
+        bool::deserialize(
+            &mut &*self.get(&paths::VALIDATOR_SET_UPDATE_COMPLETED)
+            .ok_or(KVGetError::ValueNotFound{key: Key::ValidatorSetUpdateComplete})?
+        ).map_err(|err| KVGetError::DeserializeValueError{key: Key::ValidatorSetUpdateComplete, source: err})
+    }
+
+    /* ↓↓↓ Validator Set State ↓↓↓ */
+
+    fn validator_set_state(&self) -> Result<ValidatorSetState, KVGetError> {
+        Ok(
+            ValidatorSetState::new(
+                self.committed_validator_set()?,
+                self.previous_validator_set()?, 
+                self.validator_set_update_block_height()?,
+                self.validator_set_update_complete()?
+            )
+        )
+    }
+
+    /* ↓↓↓ Highest View Voted ↓↓↓ */
+    
+    fn highest_view_voted(&self) -> Result<Option<ViewNumber>, KVGetError> {
+        if let Some(bytes) = self.get(&paths::HIGHEST_VIEW_VOTED) {
+            let view_number = ViewNumber::deserialize(&mut &*bytes)
+            .map_err(|err| KVGetError::DeserializeValueError{key: Key::HighestViewVoted, source: err})?;
+            Ok(Some(view_number))
+        } else {
+            Ok(None)
+        }
+    }
+
+}
+
+// TODO: impl Display, or import via thiserror
+#[derive(Debug)]
+pub enum KVGetError {
+    DeserializeValueError{key: Key, source: std::io::Error},
+    ValueNotFound{key: Key},
```

Comment:
Inconsistency in naming. For `BlockTreeError` the analogous variant is `BlockExpectedButNotFound`, so should this be `ValueExpectedButNotFound`?

Reply:
Fair, though this is because I thought Block Tree is more high level, so that's where the expectations about its contents come into play. 

But I can rename it. Or alternatively rename `BlockExpectedButNotFound` to `BlockNotFound` for consistency and conciseness.